### PR TITLE
[PluginSystem] Python 3 fixes

### PR DESCRIPTION
**Pull request by foundrytom**
Some minor tweaks to the `PluginSystem` so that it works better in Python 3. Adds test coverage to verify fixes.

### CODE REVIEW

File: tests/openassetio/pluginSystem/test_pluginsystemmanagerfactory.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from openassetio.logging import LoggerInterface
+from openassetio.pluginSystem import PluginSystemManagerFactory
+
+
+@pytest.fixture
+def a_logger():
+    return mock.create_autospec(spec=LoggerInterface)
+
+
+class Test_PluginSystemManagerFactory:
```
Comment (feltech) [inline tests/openassetio/pluginSystem/test_pluginsystemmanagerfactory.py:40]: [consider] Based on CONTRIBUTING this should perhaps be two classes, e.g.

```python
class Test_PluginSystemManagerFactory_init:
    def test_when_no_plugin_path_then_logs_warning(...):
        ...

class Test_PluginSystemManagerFactory_identifiers:
    def test_when_no_plugin_path_then_returns_empty_list(...):
        ...

    def test_when_empty_plugin_path_then_returns_empty_list(...):
        ...
```

**Review by feltech — APPROVED** [review]

File: tests/openassetio/pluginSystem/test_pluginsystemmanagerfactory.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""
+These tests check the functionality of the plugin system.
+"""
+
+# pylint: disable=no-self-use
```
[the remark this answers is not in the document]
Comment (foundrytom) [reply tests/openassetio/pluginSystem/test_pluginsystemmanagerfactory.py:20]: Cleaned up [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/127/commits/225d37412d6670341ebfff17348da655abcf01df).

[the remark this answers is not in the document]
Comment (foundrytom) [reply tests/openassetio/pluginSystem/test_pluginsystemmanagerfactory.py:20]: Good call, restructured tests [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/127/commits/7ee8601e2dcd5f39195f17d0ac24441edb31e195).

**Review by feltech — APPROVED** [review]
Nice, squash and merge when ready.

**Pull request merged**